GPflowOpt/bo.py

```diff
@@ -86,6 +117,8 @@ def _update_model_data(self, newX, newY):
         assert self.acquisition.data[0].shape[1] == newX.shape[-1]
         assert self.acquisition.data[1].shape[1] == newY.shape[-1]
         assert newX.shape[0] == newY.shape[0]
+        if newX.size == 0:
```

Comment:
will this ever happen? As far as I know we cant empty GPflow models so data[0] will never be empty.

Reply:
this line avoids _needs_setup = True in case i.e. the EmptyDesign is configured as initial design (as is by default)

Reply:
As a sidenote, as GPflow doesn't support models with no data I actually see no use case for BOptimizer having an initial design parameter.